pycloudlib/gce/cloud.py

```diff
@@ -27,28 +27,35 @@ class GCE(BaseCloud):
     _type = 'gce'
 
     def __init__(
-        self, tag, timestamp_suffix=True, project=None, region=None, zone=None
+        self, tag, timestamp_suffix=True, credentials_path=None, project=None,
+        region=None, zone=None
     ):
         """Initialize the connection to GCE.
 
         Args:
             tag: string used to name and tag resources with
             timestamp_suffix: bool set True to append a timestamp suffix to the
                 tag
+            credentials_path: path to credentials file for GCE
             project: GCE project
             region: GCE region
             zone: GCE zone
         """
         super().__init__(tag, timestamp_suffix)
         self._log.debug('logging into GCE')
 
+        if not all([credentials_path, project, region, zone]):
+            raise ValueError(
+                'credentials_path, project, region, and zone are required!')
+
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(credentials_path)
+        os.environ["GOOGLE_CLOUD_PROJECT"] = str(project)
+
```

Comment:
With this we are mandating that the optional parameters are required.
I think instead we should "export" the params that are provided with something like the following:

```suggestion
if credentials_path:
    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(credentials_path)
if project:
    os.environ["GOOGLE_CLOUD_PROJECT"] = str(project)
```